twilio/twilio.d.ts

```diff
@@ -0,0 +1,1408 @@
+    list: RestMethod;
+  }
+
+  export interface ServiceChannelInstance {
+    get: RestMethod;
+    post: RestMethod;
+    delete: RestMethod;
+    update: RestMethod;
+
+    messages: ServiceChannelMessageResource;
+    members: ServiceChannelMemberResource;
+  }
+
+  export interface ServiceChannelResource {
+    (resourceSid: string): ServiceChannelInstance;
+
+    get: RestMethod;
+    post: RestMethod;
+    create: RestMethod;
+    list: RestMethod;
+  }
+
+  export interface ServiceInstance {
+    get: RestMethod;
+    post: RestMethod;
+    delete: RestMethod;
+    update: RestMethod;
+
+    users: ServiceUserResource;
+    roles: ServiceRoleResource;
+    channels: ServiceChannelResource;
+  }
+
+  export interface ServiceResource {
+    (resourceSid: string): ServiceInstance;
+
+    get: RestMethod;
+    post: RestMethod;
+    create: RestMethod;
+    list: RestMethod;
+  }
+
+  /// resources/lookups/PhoneNumbers.js
+  export interface PhoneNumberInstance {
+    get: RestMethod;
+  }
+
+  export interface PhoneNumberResource {
+    (resourceSid: string): PhoneNumberInstance;
+  }
+
+  /// resources/monitor/Alerts.js
+  export interface AlertInstance {
+    get: RestMethod;
+  }
+
+  export interface AlertResource {
+    (resourceSid: string): AlertInstance;
+
+    get: RestMethod;
+    list: RestMethod;
+  }
+
+  /// resources/monitor/Events.js
+  export interface EventInstance {
+    get: RestMethod;
+  }
+
+  export interface EventResource {
+    (resourceSid: string): EventInstance;
+
+    get: RestMethod;
+    list: RestMethod;
+  }
+
+  /// resources/pricing/Messaging.js
+  export interface CountryInstance {
+    get: RestMethod;
+  }
+
+  export interface CountryResource {
+    (resourceSid: string): CountryInstance;
+
+    get: RestMethod;
+    list: RestMethod;
+  }
+
+  export interface MessagingResource {
+    countries: CountryResource;
+  }
+
+  /// resources/pricing/PhoneNumbers.js
+  export interface PhoneNumberResource {
+    countries: CountryResource;
+  }
+
+  /// resources/pricing/Voice.js
+  export interface NumberInstance {
+    get: RestMethod;
+  }
+
+  export interface NumberResource {
+    (resourceSid: string): NumberInstance;
+
+    get: RestMethod;
+    list: RestMethod;
+  }
+
+  export interface VoiceResource {
+    countries: CountryResource;
+    numbers: NumberResource;
+  }
+
+  /// resources/sip/CredentialLists.js
+  export interface CredentialListInstance {
+    credentials: CredentialResource;
+    
+    get: RestMethod;
+    post: RestMethod;
+    delete: RestMethod;
+    update: RestMethod;
+  }
+
+  export interface CredentialListResource {
+    (resourceSid: string): CredentialListInstance;
+
+    get: RestMethod;
+    post: RestMethod;
+    create: RestMethod;
+    list: RestMethod;
+  }
+
+  /// resources/sip/Domains.js
+  export interface IPAccessControlListMappingInstance {
+    get: RestMethod;
+    delete: RestMethod;
+  }
+
+  export interface IPAccessControlListMappingResource {
```

Comment:
seems like these could be a single `ListMappingResource<T>` and then have a bunch of type aliases like:

```ts
type IPAccessControlListMappingInstance = ListMappingResource<IPAccessControlMappingInstance>;
type CredentialListResource = ListMappingResource<CredentialListInstance>;
...
```